danascheider/skyrim_inventory_management_frontend
# Enable user to edit inventory list title from UI

## Context

[**Enable user to edit inventory list title from UI**](https://trello.com/c/gyUeCoae/147-enable-user-to-edit-inventory-list-title-from-ui)

Continuing work on the UI for inventory lists, we need a way to edit the title of inventory lists from the UI. We'll use the same approach as editing shopping lists, reusing the `ShoppingListEditForm` component

## Changes

* Rename `ShoppingListEditForm` to `ListEditForm` since it is now reused for multiple list types
* Add list title editing functionality to the `InventoryList` component
* Add `updateInventoryList` function to the SIM API to make `PATCH` requests for inventory lists
* Add `performInventoryListUpdate` function to the `InventoryListsContext` to make update requests and handle the response
* Add handlers to Storybook stories for the `InventoryList` and `InventoryPage` components
* Add Jest tests for updating an inventory list title
* Update docs

## Considerations

We're continuing with the same approach we took to shopping lists. Before considering this epic done, we'll do a refactor to see how we can DRY up some duplicated code.

## Manual Test Cases

Test editing the title of an inventory list, both setting it to a new, non-empty value and an empty value. The non-empty value should be titlecased and immediately appear as the title of the list, hiding the form. The empty value should be replaced by a title like "My List 1" when the API response finishes.

You'll need to account for the following responses from the API:

* 200 (successful update)
* 401 (wait to get logged out or program your API to always return a 401)
* 404 (either remove a list on another device before editing it or program your API to always return a 404)
* 422 (changing the name of the list to "All Items" works for this)
* 500 (program your API to always return a 500 error)

## Screenshots and GIFs

### Editable Inventory List with Form Visible

<img width="779" alt="Screen Shot 2021-08-15 at 11 29 31 am" src="https://user-images.githubusercontent.com/5115928/129464307-6c3d1749-41f1-4b01-9d52-514e5b36a1c0.png">

### Inventory Lists Page Showing Editable and Non-Editable Lists

<img width="1026" alt="Screen Shot 2021-08-15 at 11 30 25 am" src="https://user-images.githubusercontent.com/5115928/129464315-2e666922-d747-45c2-a2ef-04ca2d0ae92e.png">
